openedx/paragon
# feat: adds OverflowScroll, useOverflowScroll, CardCarousel

## Description

Adds the following:
* Documentation page: `useIsVisible`
* React hook (with documentation): `useOverflowScroll`
* React hook: `useOverflowScrollItems`
* React component (with documentation): `OverflowScroll`
  * React context: `OverflowScrollContext`
  * React component: `OverflowScroll.StartSentinel`
  * React component: `OverflowScroll.EndSentinel`
  * React component: `OverflowScroll.Items`
* React component: `CardCarousel` 

Updates the following:
* `CardDeck`
* Shows example on documentation of using `OverflowScroll` to build a `Chip` carousel (in the future, there might be a standalone `ChipCarousel` component, but this is a start).

**Note: tests are still a WIP.**

### Testing Instructions
1. Checkout the `ags/overflowscroll-cardcarousel` branch locally.
2. Run `npm i` to install top-level dependencies.
3. Run `cd www && npm i` to install dependencies for the documentation website.
4. Run `npm start` to start the documentation website.
5. Navigate to and verify the following:
  * http://localhost:8000/components/overflowscroll/overflowscroll/
  * http://localhost:8000/components/overflowscroll/useoverflowscroll/
  * http://localhost:8000/components/card/#carddeck
  * http://localhost:8000/components/card/#cardcarousel
  * http://localhost:8000/components/chip/#chipcarousel

### Deploy Preview

https://deploy-preview-1884--paragon-openedx.netlify.app/

## Merge Checklist


## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
